### PR TITLE
Add auto-publishing of new tags

### LINKED ISSUE #303: Automated release
# Problem / Motivation

8h ago, a feature I was highly anticipating was [merged](https://github.com/zeit/next.js/pull/301) into the `master` branch. Unfortunately, I will have to wait for someone to manually release this feature, in order to be able to use it without to much hassle.

# Solution

> Everything that can be automated will be automated.
> \- [Zuboff's 1st Law](https://github.com/semantic-release/semantic-release)

If we believe Zuboff, releasing could and therefor will be automated. Ideally using [semantic-release](https://github.com/semantic-release/semantic-release). This code runs on a CI server and inspects the commit messages of a merged PR. Following [semantic versioning](http://semver.org/) and a convention on commit messages, `semantic-release` chooses what the next version number is and triggers a releases. Commit messages are used to create the change log.

Did you ever consider automated releasing? Is there any reason we should not use it with `next.js`?

### PR DESCRIPTION
**Pull request by timneutkens**
cc @leo 

Fixes #303

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,57 @@
...
+          "HGFqez0XKG3vWJUmDjC1veVJmdASUYviaAbJCqQi4JbgNs49AelhYk3w4vl7K6wOnS2OUPD7tICKNDpxBhunKmVlEkLQtqSnQZjmi4x2tGufGTka0c9O7qNSFsiQXAzta/97MnhmGnJocFsppNNsS21H2aswNQJo+P2QXvmHvFbwzda6k3NwtOc43K+0rW1NM164UwRjtD38LWF+jvhYyysTrWTFXbYUp2o1Q8kI31pn4pMhhOP3qOnuS1jpOzjk6uZ4g5u4cMeK2eSZ/j8B1PNUBXjBfu75IU0yzvlr3ypN7hYcutDTBn4ISDfFM874FdrFhcGiMy7gMO/iwC5LTUqkBGYJT6sYdWEkKU7HQygKeFkinrkmypFdJf9ufw2EDjVCUDYnC5JbnOwu4d3vFHcg3GU6Oa7g1nrScPnZ+jkoM4ugia/3vhFO7serq4qXEj7Mv5p78mW41gxHUTgf8uqwhiOnqo9Or7UsR7vU/hB3uU0DY/nq5BI3Wt4XANrlTDeMDExjDACIQ1GU089va88NbWyU1XDD2zs+o1Y6OigAkn/yZ8V/z52eYcA5znTxcb4s8P36pKCSEkxyRtpHNf138hDx9qC2Hze1alBDfUC4iRhiZCQN4CXhSxVdb6HjZHNwCyWCO16LZLbnBvCbIMR+0WjnL0KzqsUTOboZHkQ="
+      },
+      skip_cleanup: true,
+      on: {
+        condition: '"$(git rev-parse --abbrev-ref HEAD)" == "canary"',
+        tags: true
+      }
+    }
+  ],
+  after_deploy: "npm install -g now"
```
Comment (leo) [inline .travis.yml]: You can remove this... :blush:

**Review by leo — CHANGES_REQUESTED** [review]
😏

File: .travis.yml (hunk 1 above)
Comment (timneutkens) [reply .travis.yml]: Fixed ❤️

**Comment by leo** [comment]
😍

**Pull request merged**